From c41fca3daed8b515c4022ab02232acab8b1ec4c4 Mon Sep 17 00:00:00 2001
From: Pavel Moravec <pmoravec@redhat.com>
Date: Tue, 20 May 2025 13:18:26 +0200
Subject: [PATCH] [dnf] Scrub passwords in repository URIs

Closes: #4018

Signed-off-by: Pavel Moravec <pmoravec@redhat.com>
---
 sos/report/plugins/dnf.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sos/report/plugins/dnf.py b/sos/report/plugins/dnf.py
index 4a1e52b9..69224955 100644
--- a/sos/report/plugins/dnf.py
+++ b/sos/report/plugins/dnf.py
@@ -150,4 +150,10 @@ class DNFPlugin(Plugin, RedHatPlugin):
         #
         self.do_file_sub("/etc/dnf/dnf.conf", regexp, repl)
 
+        # Scrub credentials in http URIs
+        self.do_paths_http_sub([
+            '/etc/yum.repos.d/*',
+            '/var/log/dnf.*',
+        ])
+
 # vim: set et ts=4 sw=4 :
-- 
2.49.0

