--- /dev/null			2025-04-03 01:35:45.132999852 +0200
+++ sos-4.9.2/bin/sosreport	2025-04-15 13:54:04.924751581 +0200
@@ -0,0 +1,5 @@
+#!/usr/bin/python3
+msg = ("sosreport binary is deprecated, use 'sos report' instead")
+print(msg)
+
+# vim:ts=4 et sw=4

--- /dev/null			2025-04-03 01:35:45.132999852 +0200
+++ sos-4.9.2/bin/sos-collector	2025-04-15 15:10:17.780281627 +0200
@@ -0,0 +1,5 @@
+#!/usr/bin/python3
+msg = ("sos-collector binary is deprecated, use 'sos collector' instead")
+print(msg)
+
+# vim:ts=4 et sw=4

--- sos-4.9.2/setup.py	2025-04-15 15:17:21.938635468 +0200
+++ sos-4.9.2/setup.py	2025-04-15 15:17:41.328198501 +0200
@@ -34,7 +34,7 @@
     maintainer_email='jacob.r.hunsaker@gmail.com',
     url='https://github.com/sosreport/sos',
     license="GPLv2+",
-    scripts=['bin/sos'],
+    scripts=['bin/sos', 'bin/sosreport', 'bin/sos-collector'],
     data_files=data_files,
     packages=find_packages(include=['sos', 'sos.*'])
 )
